### PR TITLE
Use gwdatafind instead of glue.datafind

### LINKED ISSUE #2916: Conda installed version pycbc_submit_dax unable to start datafind connection
This error was found during running of pycbc_submit_dax, but can be reduced to the following:

```
python -c "from pycbc.frame import datafind_connection; server=datafind_connection()"
basename: missing operand
Try 'basename --help' for more information.
basename: missing operand
Try 'basename --help' for more information.
basename: missing operand
Try 'basename --help' for more information.
basename: missing operand
Try 'basename --help' for more information.
basename: missing operand
Try 'basename --help' for more information.
basename: missing operand
Try 'basename --help' for more information.
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/paul.hopkins/.conda/envs/pycbc_only/lib/python3.7/site-packages/pycbc/frame/frame.py", line 289, in datafind_connection
    host=server, port=port, cert_file=cert_file, key_file=key_file)
  File "/home/paul.hopkins/.conda/envs/pycbc_only/lib/python3.7/site-packages/glue/datafind.py", line 414, in __init__
    http_client.HTTPSConnection.__init__(self, host, **kwargs)
  File "/home/paul.hopkins/.conda/envs/pycbc_only/lib/python3.7/http/client.py", line 1358, in __init__
    blocksize=blocksize)
TypeError: __init__() takes from 1 to 2 positional arguments but 5 were given
```

Also this shows lots of errors from basename, which I did not see before.

### PR DESCRIPTION
**Pull request by duncanmmacleod**
This PR fixes #2916 by patching `pycbc.frame.frame` to use `gwdatafind`, rather than the (long deprecated) `glue.datafind` module.

### CODE REVIEW

**Comment by titodalcanton** [comment]
Looks like this is introducing an error in the workflow test:
```
Traceback (most recent call last):

  File "/cvmfs/oasis.opensciencegrid.org/ligo/sw/pycbc/x86_64_rhel_7/virtualenv/pycbc-master/bin/pycbc_make_coinc_search_workflow", line 131, in <module>

    seg_file=science_seg_file)

  File "/cvmfs/oasis.opensciencegrid.org/ligo/sw/pycbc/x86_64_rhel_7/virtualenv/pycbc-master/lib/python2.7/site-packages/pycbc/workflow/datafind.py", line 144, in setup_datafind_workflow

    outputDir, tags=tags)

  File "/cvmfs/oasis.opensciencegrid.org/ligo/sw/pycbc/x86_64_rhel_7/virtualenv/pycbc-master/lib/python2.7/site-packages/pycbc/workflow/datafind.py", line 562, in setup_datafind_runtime_frames_single_call_perifo

    outputDir, tags=tags)

  File "/cvmfs/oasis.opensciencegrid.org/ligo/sw/pycbc/x86_64_rhel_7/virtualenv/pycbc-master/lib/python2.7/site-packages/pycbc/workflow/datafind.py", line 512, in setup_datafind_runtime_cache_single_call_perifo

    endTime, ifo, tags=tags)

  File "/cvmfs/oasis.opensciencegrid.org/ligo/sw/pycbc/x86_64_rhel_7/virtualenv/pycbc-master/lib/python2.7/site-packages/pycbc/workflow/datafind.py", line 964, in run_datafind_instance

    dfCache.ifo = ifo

AttributeError: 'list' object has no attribute 'ifo'
```
@duncanmmacleod anything obvious to you? It looks like `connection.find_frame_urls()` is now returning a list, which does not like being given an `ifo` attribute.

**Comment by tdent** [comment]
@duncanmmacleod for this to proceed, it requires reviewer(s) to be assigned.

File: pycbc/workflow/datafind.py [hunk 1]
```diff
@@ -952,10 +952,12 @@ def run_datafind_instance(cp, outputDir, connection, observatory, frameType,
     # directory to check if this was expected.
     log_datafind_command(observatory, frameType, startTime, endTime,
                          os.path.join(outputDir,'logs'), **dfKwargs)
-    logging.info("Asking datafind server for frames.")
-    dfCache = connection.find_frame_urls(observatory, frameType,
-                                        startTime, endTime, **dfKwargs)
-    logging.info("Frames returned")
+    logging.debug("Asking datafind server for frames.")
```
Comment (titodalcanton) [inline pycbc/workflow/datafind.py]: I suggest keeping this to `logging.info` due to a recent change in default logging level which would make debug logging invisible.

**Comment by titodalcanton** [comment]
This looks good to me, but since it is adding a new depencency, I would like to hear @spxiwh's and @ahnitz's opinions as well.

File: pycbc/frame/frame.py [hunk 2]
```diff
@@ -259,38 +260,7 @@ def datafind_connection(server=None):
...
-
-    # Open connection to the datafind server
-    if cert_file and key_file:
-        connection = glue.datafind.GWDataFindHTTPSConnection(
-                host=server, port=port, cert_file=cert_file, key_file=key_file)
-    else:
-        connection = glue.datafind.GWDataFindHTTPConnection(
-                host=server, port=port)
-    return connection
+    return gwdatafind.connect(host=server)
```
Comment (duncan-brown) [inline pycbc/frame/frame.py]: Does gwdatafind internally inspect `LIGO_DATAFIND_SERVER` if server is None?

Comment (duncanmmacleod) [reply pycbc/frame/frame.py]: Yes, https://git.ligo.org/gwdatafind/gwdatafind/blob/3caa19d256ad20a46e09dafa8ed468b4f90539e4/gwdatafind/ui.py#L49.

**Pull request merged**